crates/runtime/tests/models/openai.rs

```diff
@@ -672,13 +672,31 @@ async fn verify_similarity_search_chat_completion(
         a_str.cmp(&b_str)
     });
 
-    // Create a combined JSON array with all filtered values
-    let combined_value = serde_json::Value::Array(stable_inputs);
+    // Instead of creating a snapshot, verify keywords contain expected values
+    let mut found_journalist = false;
+    let mut found_vehicle = false;
+
+    for input in &stable_inputs {
+        if let Some(keywords) = input.get("keywords").and_then(|k| k.as_array()) {
+            for keyword in keywords {
+                if let Some(keyword_str) = keyword.as_str() {
+                    if keyword_str.contains("journalist") {
+                        found_journalist = true;
+                    } else if keyword_str.contains("vehicle") {
+                        found_vehicle = true;
+                    }
+                }
+            }
+        }
+    }
 
-    // Verify Task History
-    insta::assert_snapshot!(
-        "chat_2_document_similarity_tasks",
-        serde_json::to_string_pretty(&combined_value).expect("Failed to serialize task_input")
+    assert!(
+        found_journalist,
+        "Expected to find 'journalists' in keywords"
+    );
+    assert!(
+        found_vehicle,
+        "Expected to find a keyword containing 'vehicle'"
```

Comment:
Put the actual inputs into the error message as well, so that its clear to understand why the test fails if it does.